presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1426,9 +1426,56 @@ void PrestoServer::populateMemAndCPUInfo() {
   });
   RECORD_METRIC_VALUE(kCounterNumQueryContexts, numContexts);
   cpuMon_.update();
+  checkOverload();
   **memoryInfo_.wlock() = std::move(memoryInfo);
 }
 
+void PrestoServer::checkOverload() {
```

Comment:
Maybe offloading all these in a separate class?

Reply:
@tanjialiang 
I would rather keep this logic here.
Next we will need Server to tell TaskManager that we are overloaded or no longer, so it can act on the Task queue.
Doing this in some another side class seems wrong.
Spinning up the class to just decide if we are overloaded based on a couple of metrics seems too much.

If that somehow grows over time, then we can refactor.

Reply:
We are adding more and more stuff to PrestoServer now. It will be easier to read and maintain if we move these logic to a separate class maybe called SystemMonitor. We can start from this one and maybe move a couple of stuff from current PrestoServer there, as well.